### PR TITLE
add hip support 

### PR DESCRIPTION
**Pull request by zhangxiao-stack**

### CODE REVIEW

**Review by tanyokwok — DISMISSED** [review]
Great, LGTM!

**Comment by tanyokwok** [comment]
@zhangxiao-stack You have to rebase the PR to avoid the conflicts.

**Comment by qiuxiafei** [comment]
@zhangxiao-stack please resolve the conflicts to unlock CI runners.

File: tao_compiler/mlir/xla/ral/context/transpose_impl.cc [hunk 1]
```diff
@@ -32,7 +32,8 @@ using tao::ral::gpu::stream_t;
 namespace tao {
 namespace ral {
 
-#if defined(GOOGLE_CUDA) || defined(TENSORFLOW_USE_ROCM)
+
+#if defined(GOOGLE_CUDA)
```
Comment (tanyokwok) [inline tao_compiler/mlir/xla/ral/context/transpose_impl.cc:36]: Nit: should not remove `defined(TENSORFLOW_USE_ROCM)`; Add a new flag if needed by DCU.

**Review by tanyokwok — APPROVED** [review]
LGTM, Thanks for this effort!

**Pull request merged**